pandas/core/groupby.py

```diff
@@ -2586,6 +2586,15 @@ def _get_grouper(obj, key=None, axis=0, level=None, sort=True,
     """
     group_axis = obj._get_axis(axis)
 
+    if key is None and level is not None and \
+       isinstance(group_axis, MultiIndex):
```

Comment:
so maybe move this down a bit (under ``level is not None``). I don't want this to be a bespoke condition. I *think* you can remove the isinstance check of MultiIndex.